AC_Moderation_Backend/controllers/buildings_controllers.js

```diff
@@ -4,14 +4,31 @@ const strings = require('../helpers/strings');
 async function postNewBuilding(req, res){
     new_building = new building();
     new_building.buildingLetter = req.body.buildingLetter;
-    new_building.subdepartments = req.body.subdepartments;
-    new_building.description = req.body.description;
+    //new_building.subdepartments = req.body.subdepartments;
+    if (req.body.description != null) new_building.description = req.body.description;
 
     const saved = await new_building.save();
     if(saved) return res.status(200).send({message: strings.Saved});
     return res.status(500).send({message: strings.err_500});
 }
 
+async function postNewArea(req, res){
+    var letter = req.body.buildingLetter;
+    var cr = req.body.classroom;
+    var parent_exists = await building.exists({buildingLetter: letter});
+    if (parent_exists) {
+        var parent_building = building.findOne({buildingLetter: letter}, (err, done)=>{
+            if(err) return null
+            return done
```

Comment:
So... We don't need to check anything, 'cause we already know that **parent_exists**.
I read in [StackOverflow](https://stackoverflow.com/questions/33383840/is-there-a-javascript-equivalent-for-the-python-pass-statement-that-does-nothing) that we can actually do nothing into an **if** statement.

```suggestion
            // Nothing to check here
```

Reply:
No, even if we don't use the if, the line of `return done` must be there. Or we could just delete the promise (is that what is called the (err, done)? ). But your suggestion isn't valid

Reply:
I see...
Humm, then, we should check what **parent_building** is giving us. If it's **null** (Internet crashing or some stuff like that) the application will explode. Let's add an additional **if** statement to verify the content of **parent_building**.

Reply:
then we should let it stay like that because that's what it does

Reply:
No, it's not. 
Whatever **parent_building** has, the code executes the line 24.